### ISSUE TITLE
Introspection or trace for warning "Action timed out"

### ISSUE THREAD

**Issue by adverley** [airo-ugent/airo-mono#45]
The current implementation warns users if actions timed out which is really useful. It might be handy to know the source of the timed-out action.

**Comment by tlpss** [comment]
Agreed. Errors (instead of warnings) would do this but I thought it would not always be desirable to raise an error. 
If we believe errrors are fine, we get the trace for free. If not we can use `inspect` to print the stack. https://stackoverflow.com/questions/1156023/print-current-call-stack-from-a-method-in-code 

@adverley what do you think? should we raise errors or warnings?
